### PR TITLE
[NNPA] Add backend test for NNPA

### PR DESCRIPTION
**Pull request by imaihal**
This PR enables backend tests for NNPA.

- Test code
The test code for NNPA is shared with that of the CPU.  `variables.py` and `common.py` are slightly modified to use `maccel` flag. Newly added CMakeLists.txt set up the test code for NNPA.
- ATOL and RTOL for NNPA
To set appropriate ATOL and RTOL for NNPA in onnx-mlir (without modifying onnx), this PR added `InferenceBackendTest` class by inheriting `onnx.backend.test.BackendTest`, and set ATOL and RTOL for NNPA by overriding `assert_similar_outputs` function. This approach is similar with signature backend test(`test/backend/signature_backend.py`).  Default ATOL and RTOL can be replaced with new one using environment variables, `TEST_ATOL` and `TEST_RTOL`. New ATOL and RTOL are set in `CMakeLists.txt` .

- Check NNPA instruction
To check if NNPA instruction is generated in shared library, this PR provides `check_instruction()` function. The environment variable `TEST_INSTRUCTION_CHECK` need to be set to true and the test case name and instruction name need to be listed in `TEST_CASE_BY_USER` such as `test_gemm_default_matrix_bias_cpu,zdnn_matmul_op`. 

Test results
```
Scanning dependencies of target check-onnx-backend-nnpa
Test info:
  temporary results are in dir:./
  targeting maccel: NNPA

  if ref_outputs[i].dtype == np.object:
.sssssssssssssssss.s.sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss
ssssss.s.s.s.s.s.s.s.s.s.s.s.s.s.s.sss.sss.sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss
ssssssssssssssssssssssssssssssssssssssss.s.s.s.sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss.s.s.sssssssssssssssssssssssssssssssssssssssssssssssssss.s.sssssssssssssssssssssssssssssssssssssssssssssssssssss.s.s.s.s.s.s.s.s.s.s.s.s.s.s.sssssssssssssss.s.s.sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss.s.sssssssss.sssssss.sssssss.sssssssssssss.s.s.s.s.s.sss.sss.s.s.s.s.s.s.s.sssssssss.s.s.s.s.s.s.s.s.s.s.sss.sssssssssssssss.sss.s.s.s.s.s.s.s.sssssssssssssssssssssssssssssssssssssss.s.s.sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss.s.s.s.s.s.s.s.sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss.sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss.sss.sss.sss.sss.sssssssssssssssssssssssssssssssssssssssssssssssssssss.s.s.sss.s.s.sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss.sss.s.s.s.sssssssssssssssssssssssssssssssssssssssssssssssss
----------------------------------------------------------------------
Ran 1916 tests 

OK (skipped=1800)
Start downloading model inception_v1 from https://s3.amazonaws.com/download.onnx/models/opset_9/inception_v1.tar.gz
Done
Start downloading model resnet50 from https://s3.amazonaws.com/download.onnx/models/opset_9/resnet50.tar.gz
Done
Start downloading model shufflenet from https://s3.amazonaws.com/download.onnx/models/opset_9/shufflenet.tar.gz
Done
Start downloading model squeezenet from https://s3.amazonaws.com/download.onnx/models/opset_9/squeezenet.tar.gz
Done
Start downloading model vgg19 from https://s3.amazonaws.com/download.onnx/models/opset_9/vgg19.tar.gz
Done
[100%] Built target check-onnx-backend-nnpa
```

### CODE REVIEW

**Review by AlexandreEichenberger — COMMENTED** [review]
I like the directory structure that you did, by adding an "accelerator" subdirectory in the top test dir. Is it working well, or is it requiring to add a lot of redundant code? I would like to have your input as we should be striving to have as little code replication as possible. Thanks

**Comment by imaihal** [comment]
@AlexandreEichenberger 

> I like the directory structure that you did, by adding an "accelerator" subdirectory in the top test dir. Is it working well, or is it requiring to add a lot of redundant code? I would like to have your input as we should be striving to have as little code replication as possible. 

I think basically we can avoid redundant code even with this structure. Currently only `CMakeLists.txt` (`test/accelerators/NNPA/backend/CMakeLists.txt`) is bit redundant, but I think it is acceptable.

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM, thanks

**Comment by imaihal** [comment]
@AlexandreEichenberger I'm trying to implement a function to check if specific instruction are included in generated shared library, which is similar with one in numerical tests (`checkInstruction()`). I found a similar way in python, but still under development. Thanks!

**Comment by imaihal** [comment]
@AlexandreEichenberger @tungld Ready for review. Thanks!

File: test/accelerators/NNPA/backend/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+  set(BACKEND_TEST_COMMAND "${Python3_EXECUTABLE}" "-m" "pytest")
+  set(BACKEND_TEST_ARGS "--forked" "-n" "$$\{NPROC:-auto\}" "-q" "--silent")
+else()
+  message(STATUS "Parallel backend tests   : OFF (install pytest-xdist to enable)")
+  set(BACKEND_TEST_COMMAND ${Python3_EXECUTABLE})
+  set(BACKEND_TEST_ARGS "")
+endif()
+
+# Followings are test cases of test_to_enable_dict in test/backend/inference_backend.py.
+# Only test cases supported by zDNN are listed.
```
Comment (AlexandreEichenberger) [inline test/accelerators/NNPA/backend/CMakeLists.txt:85]: Do you mind explaining the format? is it paris of <backend-test> <zdnn-op-being-testted>

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM

File: test/accelerators/NNPA/backend/CMakeLists.txt (hunk 1 above)
Comment (imaihal) [reply test/accelerators/NNPA/backend/CMakeLists.txt]: Thanks for your comment. I added a sentence to explain. Also I updated [AccelNNPAHowToUseAndTest.md](https://github.com/imaihal/onnx-mlir/blob/nnpa_backendtest/docs/AccelNNPAHowToUseAndTest.md) and [Testing.md](https://github.com/imaihal/onnx-mlir/blob/nnpa_backendtest/docs/Testing.md) to explain the environment variables.

**Pull request merged**